File: soln2017f/Perfect.java
import java.io.*;
import java.util.*;

public class Perfect {
  public static void main(String[] args) throws IOException {

      Scanner sc = new Scanner(new FileReader("PerfectIN.txt"));
      while (sc.hasNext()) {
        System.out.println(isPerfect(sc.nextInt()));
      }

  }

  public static boolean isPerfect(int n) {
    if (n <= 0)
      return false;

    int sum = 0;

    for (int i = 1; i <= Math.sqrt(n); i++) {
      if (n % i == 0) {
        sum += i;
        if (i * i != n)
          sum += n / i;
      }
    }

    return (sum - n) == n;
  }
}

File: soln2017f/Perfect.py
import math

def is_perfect(n):
    if n <= 0:
        return False

    sum = 0

    for i in range(1, int(math.ceil(math.sqrt(n)))):
        if n % i == 0:
            sum += i
            if i * i != n:
                sum += n / i

    return (sum - n) == n

if __name__ == '__main__':
    with open('PerfectIN.txt', 'r') as f:
        while True:
            n = f.readline()
            if n == '':
                break
            n = int(n)
            out = is_perfect(n)
            if out:
                print('true')
            else:
                print('false')
